core/error/error_list.h

```diff
@@ -93,6 +93,7 @@ enum Error {
 	ERR_HELP, ///< user requested help!!
 	ERR_BUG, ///< a bug in the software certainly happened, due to a double check failing or unexpected behavior.
 	ERR_PRINTER_ON_FIRE, /// the parallel port printer is engulfed in flames
+	ERR_PARSE_ERROR_INVALID_SURROGATE,
```

Comment:
This seems like a highly specific error to include in the main `Error` enum.
Maybe this should be handled ad hoc in the String class with a dedicated `ParseError` enum, paving the way for better error reporting on invalid strings.

Reply:
Forgive me, but is there perhaps another way to interpret what the error is? Say, instead of **ERR_PARSE_ERROR_INVALID_SURROGATE**, it could be any other of the existing ones. It... sounds hacky, but if there's a fairly close alternative, that could be it.